cmd/istioctl/register.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	"istio.io/pilot/platform/kube"
+)
+
+var (
+	registerCmd = &cobra.Command{
+		Use:   "register <svcname> <ip> [name1:]port1 [name2:]port2 ...",
```

Comment:
once again, please change to kube-register. There is no kubernetes style static registration of an in-cluster endpoint in mesos, docker swarm, Consul/Nomad, Eureka. They all come with requirements on heartbeats, health checks, and ability to dynamically adapt to conditions.

Reply:
added -platform